users/views.py

```diff
@@ -1,24 +1,96 @@
 """
 This module contains
 """
-from django.contrib.auth import get_user_model
+from django.contrib.auth import authenticate, get_user_model, login
 
 # pylint: disable= no-self-use, no-member
 from django.contrib.auth.hashers import check_password
 from django.core.exceptions import ValidationError
 from django.http import Http404
-from django.shortcuts import get_object_or_404
+from django.shortcuts import get_object_or_404, redirect, render
 from rest_framework import status
 from rest_framework.authtoken.models import Token
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
+from .contants import HOME_PAGE_URL, LOGIN, REGISTER
 from .permissions import IsAdmin
 from .serializers import UserSerializer
 
 User = get_user_model()
 
 
+class TemplateUserLogin(APIView):
+    """
+    Displays login page and allows user to login
+    """
+
+    def post(self, request):
+        """
+        Validates and logs a user in
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (render or redirect):   redirects to homepage on success
+                                    otherwise renders the login page
+        """
+        context = {}
+        try:
+            user = authenticate(
+                request,
+                username=request.POST["email"],
+                password=request.POST["password"],
+            )
+            if user:
+                login(request, user)
+                return redirect(HOME_PAGE_URL)
+            context["error_message"] = "Invalid Credentials"
+            return render(request, LOGIN, context)
+        except KeyError:
+            context["error_message"] = "There was an error logging in."
```

Comment:
We try to intimate the user about a possible solutions to something when showing an error.
```suggestion
            context["error_message"] = "There was an error logging in, please try again."
```

Reply:
Done